### ISSUE TITLE
Nick change replies and the nick from a previous session are not relayed to the UI

### ISSUE THREAD

**Issue by N00bySumairu** [CylonicRaider/euphoria-app#10]
The app's UI is not informed about the server's input about the nick of the user of the app.
This is an issue as it allows for the user to see a nick in the input bar that the server might not have accepted. It also prevents the input bar from being populated with a nick from a previous session.
The connections all respond correctly to the server side updates of the users nick, and as such you can immediately send messages with the nick from the last session if you've enable session continuation. Just the input bar does not reflect any server side input.
This should be fixed as both #1 and #9 rely on this to work.

**Comment by CylonicRaider** [comment]
Hmm... It *ought* to happen... but, then, it apparently doesn't.

**Comment by CylonicRaider** [comment]
Lessee...

1. `nick-reply` messages are handled in `EuphoriaWebSocketClient::onMessage`.
2. `ConnectionImpl::submitEvent` forwards `NickChangeEvent`s to listeners. The docs of `ConnectionListener::onNickChange` mention that the method is also called for changes of one's own nick.
3. `RoomController`'s anonymous `ConnectionListener` forwards the `NickChangeEvent` to `RoomUI::showNicks`. The latter method's docs no longer explicitly mention changes of the own nick; however, as there is no other means of inputting changes of the own nickname, it seems implied that that method is meant for it.
4. Finally, `RoomUIImpl::showNicks` only hands off the nickname changes to the user list's model without taking care to update the own nickname.

On the other hand, it does indeed look like the input bar's nickname is not *set* anywhere.

In summary, it should happen, but it — indeed — doesn't.

**Comment by CylonicRaider** [comment]
Furthermore, the room UI does not even track the current user's identity — the current nickname notwithstanding.
